pkg/controller/migmigration/postValidate.go

```diff
@@ -0,0 +1,298 @@
+package migmigration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// EventDisplay is a structure for displaying occured events
+type EventDisplay struct {
+	Pod           string    `json:"pod"`
+	LastTimestamp time.Time `json:"lastTimestamp,omitempty"`
+	Message       string    `json:"message,omitempty"`
+	Reason        string    `json:"reason,omitempty"`
+}
+
+const (
+	// ValidationFailed is a failed validation state
+	ValidationFailed = iota
+	// ValidationProgressing is a in-progress validation state
+	ValidationProgressing
+	// ValidationSucceeded is a successfull validation state
+	ValidationSucceeded
+)
+
+var podManagingResources = []schema.GroupVersionKind{
+	schema.GroupVersionKind{
+		Group: "apps",
+		Kind:  "statefulset",
+	},
+	schema.GroupVersionKind{
+		Group: "apps",
+		Kind:  "deployment",
+	},
+	schema.GroupVersionKind{
+		Group: "extensions",
+		Kind:  "daemonset",
+	},
+	schema.GroupVersionKind{
+		Group: "extensions",
+		Kind:  "replicaset",
+	},
+	schema.GroupVersionKind{
+		Group: "apps.openshift.io",
+		Kind:  "deploymentconfig",
+	},
+}
+
+func involvedObjectMatch(event corev1.Event, objectNames []string) bool {
+	for _, objectName := range objectNames {
+		if objectName == event.InvolvedObject.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// collectRelatedEvents will extract all warning events
+// which occured in connection to the resource
+func collectRelatedEvents(
+	objectNames []string,
+	eventList []corev1.Event) []EventDisplay {
+
+	filteredEventMessages := []EventDisplay{}
+
+	for _, event := range eventList {
+		if !involvedObjectMatch(event, objectNames) || event.Type != corev1.EventTypeWarning {
```

Comment:
Put the `event.Type != corev1.EventTypeWarning` first in the `||` for efficiency.